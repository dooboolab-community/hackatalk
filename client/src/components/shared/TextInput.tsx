import React, { Ref, forwardRef, useState } from 'react';
import { ReturnKeyTypeOptions, TextStyle, ViewStyle } from 'react-native';

import { Ionicons } from '@expo/vector-icons';
import styled from 'styled-components/native';

const StyledLabelText = styled.Text<{ focused: boolean }>`
  color: ${({ focused, theme }): string =>
    focused ? theme.focused : theme.inactiveColor};
  margin-bottom: 8px;
  font-size: 12px;
`;

const WrapperView = styled.View`
  align-self: stretch;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
`;

const StyledTextInputContainer = styled.View<{ isFocused?: boolean }>`
  flex-direction: row;
  justify-content: center;
  align-items: center;
  border-width: 1px;
  border-color: ${({ isFocused, theme }): string =>
    isFocused ? theme.focused : theme.lineColor};
  border-radius: 3px;
  margin-bottom: 8px;
`;

const StyledTextInput = styled.TextInput.attrs<{ focused: boolean }>(
  ({ theme, focused }) => ({
    placeholderTextColor: focused
      ? theme.placeholderFocused
      : theme.placeholder,
  }),
)<{ focused: boolean }>`
  flex: 1;
  align-self: stretch;
  color: ${({ theme }): string => theme.fontColor};
  font-size: 16px;
  padding: 22px 20px;
  border: none;
  flex-direction: row;
  align-items: center;
  justify-content: flex-start;
`;

const StyledStatusMark = styled(Ionicons).attrs(({ theme }) => ({
  name: 'md-checkmark',
  size: 24,
  color: theme.status,
}))`
  padding-right: 5%;
`;

interface Props {
  testID?: string;
  style?: ViewStyle;
  inputStyle?: TextStyle;
  isPassword?: boolean;
  multiline?: boolean;
  txtLabel?: string;
  txtHint?: string;
  txt?: string;
  onFocus?: () => void;
  onTextChanged?: (text: string) => void;
  onSubmitEditing?: (text: string) => void;
  returnKeyType?: ReturnKeyTypeOptions;
  error?: string;
}

function Shared(props: Props, ref: Ref<any>): React.ReactElement {
  const [focused, setFocused] = useState(false);

  const renderTxtLabel = (): React.ReactElement | null => {
    if (props.txtLabel)
      return (
        <StyledLabelText focused={focused}>{props.txtLabel}</StyledLabelText>
      );

    return null;
  };

  return (
    <WrapperView>
      {renderTxtLabel()}
      <StyledTextInputContainer isFocused={focused}>
        <StyledTextInput
          ref={ref}
          testID={props.testID}
          focused={focused}
          underlineColorAndroid="transparent" // android fix
          autoCapitalize="none"
          autoCorrect={false}
          multiline={props.multiline}
          onChangeText={props.onTextChanged}
          value={props.txt}
          onFocus={(): void => setFocused(true)}
          onBlur={(): void => setFocused(false)}
          placeholder={props.txtHint}
          // onSubmitEditing={this.props.onSubmitEditing}
          returnKeyType={props.returnKeyType}
          secureTextEntry={props.isPassword}
          textContentType="none" // to disable autoifill on iOS
        />
        {!!props.txt && !props.error && <StyledStatusMark />}
      </StyledTextInputContainer>
    </WrapperView>
  );
}

export default forwardRef(Shared);
