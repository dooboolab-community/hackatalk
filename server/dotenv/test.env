JWT_SECRET=dooboolab
DATABASE_URL="postgresql://test:test!@localhost:5432/postgres?schema=test"
