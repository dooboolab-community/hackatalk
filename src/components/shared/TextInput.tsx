import React, { useState } from 'react';
import { ReturnKeyTypeOptions, TextStyle, ViewStyle } from 'react-native';

import styled from 'styled-components/native';

const StyledLabelText = styled.Text<{ focused: boolean }>`
  color: ${({ focused, theme }): string =>
    focused ? theme.primary : theme.inactiveColor};
  margin-bottom: 8px;
  font-size: 12px;
`;

const StyledTextInput = styled.TextInput<{ focused: boolean }>`
  margin-bottom: 8px;
  align-self: stretch;
  color: ${({ theme }): string => theme.fontColor};
  font-size: 16px;
  padding: 22px 20px;
  border-width: 1px;
  border-color: ${({ focused, theme }): string =>
    focused ? theme.primary : theme.lineColor};
  flex-direction: row;
  align-items: center;
  justify-content: flex-start;
`;

const StyledWrapperView = styled.View`
  align-self: stretch;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
`;

interface Props {
  testID?: string;
  style?: ViewStyle;
  placeholderTextColor?: string;
  inputStyle?: TextStyle;
  isPassword?: boolean;
  multiline?: boolean;
  txtLabel?: string;
  txtHint?: string;
  txt?: string;
  onFocus?: () => void;
  onTextChanged?: (text: string) => void;
  onSubmitEditing?: (text: string) => void;
  returnKeyType?: ReturnKeyTypeOptions;
}

function Shared(props: Props): React.ReactElement {
  const [focused, setFocused] = useState(false);

  const renderTxtLabel = (): React.ReactElement | null => {
    if (props.txtLabel) {
      return (
        <StyledLabelText focused={focused}>{props.txtLabel}</StyledLabelText>
      );
    }
    return null;
  };

  return (
    <StyledWrapperView>
      {renderTxtLabel()}
      <StyledTextInput
        testID={props.testID}
        focused={focused}
        underlineColorAndroid="transparent" // android fix
        autoCapitalize="none"
        autoCorrect={false}
        multiline={props.multiline}
        onChangeText={props.onTextChanged}
        value={props.txt}
        onFocus={(): void => setFocused(true)}
        onBlur={(): void => setFocused(false)}
        placeholder={props.txtHint}
        placeholderTextColor={props.placeholderTextColor}
        // onSubmitEditing={this.props.onSubmitEditing}
        returnKeyType={props.returnKeyType}
        secureTextEntry={props.isPassword}
      />
    </StyledWrapperView>
  );
}

export default Shared;
