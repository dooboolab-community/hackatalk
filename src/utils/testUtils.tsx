import React, { ReactElement } from 'react';

import RootProvider from '../providers';
import { ThemeType } from '../types';

export const createTestElement = (
  child: ReactElement,
  themeType?: ThemeType,
): ReactElement => (
  <RootProvider initialThemeType={themeType}>{child}</RootProvider>
);

export const createTestProps = (
  obj: object = {},
  moreScreenProps?: object,
): object | unknown | any => ({
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    resetRoot: jest.fn(),
  },
  screenProps: {
    changeThemeType: jest.fn(),
    ...moreScreenProps,
  },
  ...obj,
});
