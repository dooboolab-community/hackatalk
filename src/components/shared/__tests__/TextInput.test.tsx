import 'react-native';

import * as React from 'react';

// Note: test renderer must be required after react-native.
import { createTestElement, createTestProps } from '../../../../test/testUtils';

import TextInput from '../TextInput';
import renderer from 'react-test-renderer';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let props: any;
let component: React.ReactElement;
// let testingLib: RenderResult;

describe('[TextInput] render', () => {
  beforeEach(() => {
    props = createTestProps();
    component = createTestElement(<TextInput {...props} />);
  });

  it('renders without crashing', () => {
    const rendered: renderer.ReactTestRendererJSON | null = renderer
      .create(component)
      .toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  // describe('interactions', () => {
  //   beforeEach(() => {
  //     testingLib = render(component);
  //   });

  //   it('should simulate onPress', () => {
  //     const btn = testingLib.queryByTestId('btn');
  //     act(() => {
  //       fireEvent.press(btn);
  //     });
  //     expect(cnt).toBe(3);
  //   });
  // });
});
