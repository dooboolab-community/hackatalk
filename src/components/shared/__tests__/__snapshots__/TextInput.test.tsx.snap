// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[TextInput] render renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "alignSelf": "stretch",
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    focused={false}
    onBlur={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "borderColor": "rgb(221,226,236)",
          "borderWidth": 1,
          "color": "black",
          "flexDirection": "row",
          "fontSize": 16,
          "justifyContent": "flex-start",
          "marginBottom": 8,
          "paddingBottom": 22,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 22,
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
